packages/cli/src/commands/start.ts

```diff
@@ -110,7 +109,7 @@ export class Start extends BaseCommand {
 				await Container.get(OrchestrationService).shutdown();
 			}
 
-			await Container.get(InternalHooks).onN8nStop();
+			Container.get(EventService).emit('instance-stopped'); // @TODO: Or should we await on this?
```

Comment:
Per comment

Reply:
IMO we could replace this event with `@OnShutdown` on the `Telemetry` class. Also I think the telemetry shutdown should happen only after everything else has been shut down. Now we are shutting down the active executions only after this, and they might trigger new events. Maybe we might want to send the `instance-stopped` event before shutting down if we want to make sure it gets sent

Reply:
> IMO we could replace this event with @OnShutdown on the Telemetry class. Also I think the telemetry shutdown should happen only after everything else has been shut down. 

Hmm that would make this the only event that does not go through the relay. Ideally `Telemetry` should contain pure telemetry logic, while `track` calls in response to events should live in the relay. Maybe we can keep the existing behavior in this PR and adjust the timing later.

> @TODO: Or should we await on this?

My motivation for this comment is that since we're no longer awaiting here but only sending an event, could it happen that we shut down before telemetry has had a chance to send the `instance-stopped` event?

Reply:
> could it happen that we shut down before telemetry has had a chance to send the instance-stopped event?

That might be possible. I don't think you can `await` on `EventEmitter` `emit`s (it doesn't return a promise). Or even if you can I think it's a bad practice as it should be more like "fire and forget". We could always separate sending of the `instance-stopped` event and "shutting down" the Telemetry. E.g. emit this event here, send the `track` based on it, and then separately do the other stuff from `trackN8nStop` in the `OnShutdown` hook.

As a side note, I think the `instance-stopped` event is not named the best way, as we can't always ensure that it actually does get sent. Also its now sent even before the instance actually stops. So something like `instance-stopping` would make more sense and send it immediately after the stop process has started. But that would be a breaking change for our telemetry.

Reply:
Thank you, please see [de9d930](https://github.com/n8n-io/n8n/pull/10305/commits/de9d93060cd37f0b8e3d83267c58300ab7a072cd)